tools/bug-report/pkg/bugreport/flags.go

```diff
@@ -0,0 +1,163 @@
+	commandTimeout, since                   time.Duration
+	gConfig                                 = &config2.BugReportConfig{}
+)
+
+func addFlags(cmd *cobra.Command, args *config2.BugReportConfig) {
+	// k8s client config
+	cmd.PersistentFlags().StringVarP(&args.KubeConfigPath, "kubeconfig", "c", "", bugReportHelpKubeconfig)
+	cmd.PersistentFlags().StringVar(&args.Context, "context", "", bugReportHelpContext)
+
+	// input config
+	cmd.PersistentFlags().StringVarP(&configFile, "filename", "f", "", bugReportHelpFilename)
+
+	// dry run
+	cmd.PersistentFlags().BoolVarP(&args.DryRun, "dry-run", "", false, bugReportHelpDryRun)
+
+	// full secrets
+	cmd.PersistentFlags().BoolVarP(&args.FullSecrets, "full-secrets", "", false, bugReportHelpFullSecrets)
+
+	// istio namespaces
+	cmd.PersistentFlags().StringVarP(&args.IstioNamespace, "istio-namespace", "n", bugReportDefaultIstioNamespace, bugReportHelpIstioNamespace)
+
+	// timeouts and max sizes
+	cmd.PersistentFlags().DurationVar(&commandTimeout, "timeout", bugReportDefaultTimeout, bugReportHelpCommandTimeout)
+	cmd.PersistentFlags().Int32Var(&args.MaxArchiveSizeMb, "max-size", bugReportDefaultMaxSizeMb, bugReportHelpMaxArchiveSizeMb)
+
+	// include / exclude specs
+	cmd.PersistentFlags().StringSliceVarP(&included, "include", "i", bugReportDefaultInclude, bugReportHelpInclude)
+	cmd.PersistentFlags().StringSliceVarP(&excluded, "exclude", "e", bugReportDefaultExclude, bugReportHelpExclude)
+
+	// log time ranges
+	cmd.PersistentFlags().StringVar(&startTime, "start-time", "", bugReportHelpStartTime)
+	cmd.PersistentFlags().StringVar(&endTime, "end-time", "", bugReportHelpEndTime)
+	cmd.PersistentFlags().DurationVar(&since, "duration", 0, bugReportHelpSince)
+
+	// log error control
+	cmd.PersistentFlags().StringSliceVar(&args.CriticalErrors, "critical-errs", nil, bugReportHelpCriticalErrors)
+	cmd.PersistentFlags().StringSliceVar(&args.IgnoredErrors, "whitelist-errs", nil, bugReportHelpIgnoredErrors)
+
+	// archive and upload control
+	cmd.PersistentFlags().StringVar(&args.GCSURL, "gcs-url", "", bugReportHelpGCSURL)
+	cmd.PersistentFlags().BoolVar(&args.UploadToGCS, "upload", false, bugReportHelpUploadToGCS)
+
+	// output/working dir
+	cmd.PersistentFlags().StringVar(&tempDir, "dir", bugReportDefaultTempDir, bugReportHelpTempDir)
+}
+
+func parseConfig() (*config2.BugReportConfig, error) {
+	if err := parseTimes(gConfig, startTime, endTime); err != nil {
+		log.Fatal(err.Error())
+	}
+	for _, s := range included {
+		ss := &config2.SelectionSpec{}
+		if err := ss.UnmarshalJSON([]byte(s)); err != nil {
+			return nil, err
+		}
+		gConfig.Include = append(gConfig.Include, ss)
+	}
+	for _, s := range excluded {
+		ss := &config2.SelectionSpec{}
+		if err := ss.UnmarshalJSON([]byte(s)); err != nil {
+			return nil, err
+		}
+		gConfig.Exclude = append(gConfig.Exclude, ss)
+	}
+
+	fileConfig := &config2.BugReportConfig{}
+	if configFile != "" {
+		b, err := ioutil.ReadFile(configFile)
```

Comment:
what is the use case for file config?

Reply:
i was initially about letting support create more complex default files as a template for users to customize. then they would do -f <support-default> --include=foo to override from there. this is probably most useful for having big lists of errors we want to ignore, without having to embed those in the binary.

Reply:
I personally don't see how this would end up working but fine with keeping it